src/data/equiv/basic.lean

```diff
@@ -512,13 +512,23 @@ def arrow_punit_equiv_punit (α : Sort*) : (α → punit.{v}) ≃ punit.{w} :=
 ⟨λ f, punit.star, λ u f, punit.star,
   λ f, by { funext x, cases f x, refl }, λ u, by { cases u, reflexivity }⟩
 
+/-- If `α` is `subsingleton` and `a : α`, then the type of dependent functions `Π (i : α), f
+i` is equivalent to `f i`. -/
+@[simps]
+def Pi_subsingleton {α} (β : α → Sort*) [subsingleton α] (a : α) : (Π a', β a') ≃ β a :=
+{ to_fun := eval a,
+  inv_fun := λ x b, cast (congr_arg β $ subsingleton.elim a b) x,
+  left_inv := λ f, funext $ λ b, by { rw subsingleton.elim b a, reflexivity },
+  right_inv := λ b, rfl }
+
+/-- If `α` has a unique term, then the type of dependent functions `Π (i : α), f
+i` is equivalent to `f unique`. -/
+def Pi_unique (α) (f : α → Sort*) [unique α] : (Π (i : α), f i) ≃ f default :=
```

Comment:
I would guess that `α` can be implicit here too, it's implied by `f` (and you have it implicit in the mul_equiv versions anyway)